lib/board.rb

```diff
@@ -1,89 +1,48 @@
 class Board
-  attr_reader :cells
-
-  def initialize
-    @cells = {
-      "A1" => Cell.new("A1"),
-      "A2" => Cell.new("A2"),
-      "A3" => Cell.new("A3"),
-      "A4" => Cell.new("A4"),
-      "B1" => Cell.new("B1"),
-      "B2" => Cell.new("B2"),
-      "B3" => Cell.new("B3"),
-      "B4" => Cell.new("B4"),
-      "C1" => Cell.new("C1"),
-      "C2" => Cell.new("C2"),
-      "C3" => Cell.new("C3"),
-      "C4" => Cell.new("C4"),
-      "D1" => Cell.new("D1"),
-      "D2" => Cell.new("D2"),
-      "D3" => Cell.new("D3"),
-      "D4" => Cell.new("D4"),
-    }
 
-  end
+  POSITIONS = ['A1', 'A2', 'A3', 'A4', 'B1', 'B2', 'B3', 'B4', 'C1', 'C2', 'C3', 'C4', 'D1', 'D2', 'D3', 'D4']
+
+  CELLS = Hash.new
+  POSITIONS.map { |position| CELLS[position] = Cell.new(position) }
 
   def valid_coordinate?(coordinate)
-      @cells.key?(coordinate)
+    CELLS.key?(coordinate)
   end
 
   def ship_length_placement(ship, coordinates)
     ship.length == coordinates.count
   end
 
   def split_coordinates(ship = nil, coordinates)
-    coordinates.map do |coordinate|
-      coordinate.split(//)
-    end
+    coordinates.map { |coordinate| coordinate.split(//) }
```

Comment:
This is just a shorter-hand version of iterating -  remove the `do` and `end`, and use curly braces (make sure the single defined instance - being `coordinate` is _inside_ the braces!)